plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/androidx/room/visitor/AndroidXRoomDaoVisitor.kt

```diff
@@ -0,0 +1,75 @@
+package io.sentry.android.gradle.instrumentation.androidx.room.visitor
+
+import io.sentry.android.gradle.instrumentation.AbstractSpanAddingMethodVisitor
+import io.sentry.android.gradle.instrumentation.SpanOperations
+import org.objectweb.asm.Label
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Opcodes
+
+class AndroidXRoomDaoVisitor(
+    private val className: String,
+    api: Int,
+    private val originalVisitor: MethodVisitor,
+    access: Int,
+    descriptor: String?
+) : AbstractSpanAddingMethodVisitor(
+    api = api,
+    originalVisitor = originalVisitor,
+    access = access,
+    descriptor = descriptor
+) {
+    private val childIfNullStatusOk = Label()
+    private val childIfNullFinallyPositive = Label()
+    private val childIfNullFinallyNegative = Label()
+    private val startSpanIfNull = Label()
+    private var finallyVisitCount = 0
+
+    override fun visitCode() {
+        super.visitCode()
+        originalVisitor.visitStartSpan(startSpanIfNull) {
+            visitLdcInsn(SpanOperations.DB_SQL_ROOM)
+            visitLdcInsn(className)
+        }
+        originalVisitor.visitLabel(startSpanIfNull)
+    }
+
+    override fun visitMethodInsn(
+        opcode: Int,
+        owner: String?,
+        name: String?,
+        descriptor: String?,
+        isInterface: Boolean
+    ) {
+        super.visitMethodInsn(opcode, owner, name, descriptor, isInterface)
+        // The backend binds the exception to the current span, via tracing without performance, so we don't need any special try/catch management.
+        if (opcode == Opcodes.INVOKEVIRTUAL) {
```

Comment:
hrm, does `INVOKEVIRTUAL` work with the `CLOSE` method? It used to be `INVOKEINTERFACE` before

Reply:
nice catch!
I updated the instrumentation to take the method type into consideration and updated the condition for the close method